File: src/main/java/net/alexwells/roomery/gui/container/RoomHolderContainer.java
package net.alexwells.roomery.gui.container;

import net.alexwells.roomery.tile.RoomHolderTileEntity;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraftforge.items.SlotItemHandler;

public class RoomHolderContainer extends ContainerBase {
    public static final int WIDTH = 176;
    public static final int HEIGHT = 114 + 6 * 18;

    public RoomHolderContainer(InventoryPlayer playerInventory, RoomHolderTileEntity tile)
    {
        super(176, 114 + 6 * 18);

        addSlotToContainer(new SlotItemHandler(
                tile.getItemHandler(),
                RoomHolderTileEntity.ROOM_CARD_SLOT,
                INVENTORY_BORDER + 1,
                INVENTORY_BORDER + NAME_SPACING + 5 * SLOT_SIZE + 1
        ));
        addPlayerInventorySlots(playerInventory);
    }
}

File: src/main/java/net/alexwells/roomery/mechanic/roomholder/RoomHolderContainer.kt
package net.alexwells.roomery.mechanic.roomholder

import net.alexwells.roomery.common.gui.ContainerBase
import net.alexwells.roomery.common.gui.GuiConstants.INVENTORY_BORDER
import net.alexwells.roomery.common.gui.GuiConstants.NAME_SPACING
import net.alexwells.roomery.common.gui.GuiConstants.SLOT_SIZE
import net.minecraft.entity.player.PlayerInventory
import net.minecraftforge.items.SlotItemHandler

class RoomHolderContainer(
    windowId: Int,
    playerInventory: PlayerInventory,
    tile: RoomHolderTileEntity
) : ContainerBase(RoomHolderContainerType, windowId, 176, 114 + 6 * 18) {
    init {
        addSlot(
            SlotItemHandler(
                tile.itemHandler,
                RoomHolderTileEntity.ROOM_CARD_SLOT,
                INVENTORY_BORDER + 1,
                INVENTORY_BORDER + NAME_SPACING + 5 * SLOT_SIZE + 1
            )
        )
        addPlayerInventorySlots(playerInventory)
    }
}
